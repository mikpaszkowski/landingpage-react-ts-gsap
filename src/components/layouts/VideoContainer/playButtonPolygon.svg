<svg width="40" height="46" viewBox="0 0 40 46" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M38 19.5359C40.6667 21.0755 40.6667 24.9245 38 26.4641L6.5 44.6506C3.83333 46.1902 0.500002 44.2657 0.500002 41.1865L0.500003 4.81346C0.500003 1.73426 3.83334 -0.190237 6.5 1.34936L38 19.5359Z" fill="#131E07"/>
</svg>
