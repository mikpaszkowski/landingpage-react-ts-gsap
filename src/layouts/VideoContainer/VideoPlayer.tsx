import * as React from "react";
import styled from "styled-components";
import { PlayButton } from "./PlayButton";
import { device } from "../../styles/responsive";
import { laptopLarge, mobileMedium } from "../../animations/videoPlayer";
import yogaImage from "../../assets/images/yoga-woman-image.png";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
gsap.registerPlugin(ScrollTrigger);

const VideoPlayerWrapper = styled.div`
  position: relative;
  margin: 0 auto;
  margin-bottom: 15rem;
  margin: 0 10rem;
  text-align: center;
  border-radius: 0.8rem;
  background: linear-gradient(
    180deg,
    rgba(0, 0, 0, 0.26) 0%,
    rgba(0, 0, 0, 0.06) 100%
  );

  @media ${device.tablet} {
    margin: 0 0 15rem 0;
    padding: 0;
    width: 100%;
    border-radius: 0.8rem;
  }
`;

const Image = styled.img`
  position: relative;
  z-index: -1;
  display: block;
  height: 100%;
  width: 100%;
  border-radius: 0.8rem;
`;

export const VideoPlayer = () => {
  const { useRef, useEffect } = React;

  const videoRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": () => laptopLarge(videoRef),
      "(min-width: 500px)": () => laptopLarge(videoRef),
      "(max-width: 499px)": () => mobileMedium(videoRef),
    });
  }, []);

  return (
    <VideoPlayerWrapper ref={videoRef}>
      <Image
        src={yogaImage}
        alt="Woman as freelance worker does yoga exersice in jumper."
      />
      <PlayButton />
    </VideoPlayerWrapper>
  );
};
