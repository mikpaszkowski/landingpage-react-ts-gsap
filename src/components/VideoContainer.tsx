import * as React from "react";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";
import { VideoPlayer } from "../components/VideoPlayer";
import { device } from "../styles/responsive";

const VideoContainerWrapper = styled.div`
  width: 100%;
  background-color: ${colors.primaryTheme};
  position: relative;
  overflow: visible;
  margin-bottom: 36.3rem;

  @media ${device.mobileMedium} {
    padding: 0 2.4rem;
    margin-bottom: 13.7rem;
  }
`;

const HeadLine = styled.h1`
  font-family: ${fontTypes.primaryFont};
  font-size: 8rem;
  padding-top: 6.4rem;
  margin: 0 30rem 4rem 30rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;

  @media ${device.mobileMedium} {
    font-size: 5rem;
    text-align: left;
    margin: 0;
    margin-bottom: 3.7rem;
  }
`;

const Text = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-align: center;
  margin: 4rem 53.5rem 5.6rem 53.5rem;

  @media ${device.mobileMedium} {
    font-size: 2rem;
    text-align: left;
    margin: 0;
    margin-bottom: 5.6rem;
  }
`;

export const VideoContainer = () => {
  return (
    <VideoContainerWrapper>
      <HeadLine>What our community is saying?</HeadLine>
      <Text>
        A mission-driven company that invest in and builds healthier living
      </Text>
      <VideoPlayer />
    </VideoContainerWrapper>
  );
};
