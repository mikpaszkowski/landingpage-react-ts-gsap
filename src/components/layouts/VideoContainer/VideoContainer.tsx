import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { VideoPlayer } from "./VideoPlayer";

import {
  laptopLargeHeadline,
  laptopLargeText,
  mobileMediumHeadline,
  mobileMediumText,
} from "../../../animations/videoContainer";
import { device } from "../../../styles/responsive";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
gsap.registerPlugin(ScrollTrigger);

const VideoContainerWrapper = styled.div`
  position: relative;
  z-index: 0;
  width: 100%;
  height: 81.1rem;
  background-color: ${({ theme }) => theme.palette.common.theme};
  position: relative;
  overflow: visible;
  margin-bottom: 36.3rem;

  @media ${device.tabletLarge} {
    height: 43.7rem;
  }

  @media ${device.tablet} {
    height: 50rem;
  }

  @media ${device.mobileLarge} {
    padding: 0 2.4rem;
    margin-bottom: 13.7rem;
    height: 49.7rem;
  }
`;

const HeadLine = styled.h1`
  font-family: ${fontTypes.primaryFont};
  font-size: 8rem;
  padding-top: 6.4rem;
  margin: 0 30rem 4rem 30rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;

  @media ${device.tabletLarge} {
    font-size: 6rem;
    text-align: center;
    margin: 0 8rem 3.7rem 8rem;
  }

  @media ${device.mobileLarge} {
    font-size: 5rem;
    text-align: left;
    margin: 0;
    margin-bottom: 3.7rem;
  }
`;

const Text = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-align: center;
  margin: 4rem 53.5rem 5.6rem 53.5rem;

  @media ${device.laptopLarge} {
    font-size: 2rem;
    text-align: center;
    margin: 0 15rem 4rem 15rem;
    margin-bottom: 7.5rem;
  }

  @media ${device.mobileLarge} {
    text-align: left;
    margin: 0;
    margin-bottom: 7.5rem;
  }
`;

export const VideoContainer = () => {
  const { useRef, useEffect } = React;

  const headlineRef = useRef<HTMLHeadingElement>(null);
  const textSectionRef = useRef<HTMLParagraphElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": function () {
        laptopLargeHeadline(headlineRef);
        laptopLargeText(textSectionRef);
      },

      "(min-width: 500px)": function () {
        laptopLargeHeadline(headlineRef);
        laptopLargeText(textSectionRef);
      },

      "(max-width: 499px)": function () {
        mobileMediumHeadline(headlineRef);
        mobileMediumText(textSectionRef);
      },
    });
  }, []);

  return (
    <VideoContainerWrapper>
      <HeadLine ref={headlineRef}>What our community is saying?</HeadLine>
      <Text ref={textSectionRef}>
        A mission-driven company that invest in and builds healthier living
      </Text>
      <VideoPlayer />
    </VideoContainerWrapper>
  );
};
