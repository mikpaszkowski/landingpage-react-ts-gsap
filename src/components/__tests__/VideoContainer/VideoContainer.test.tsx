import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { VideoContainer } from "../../layouts/VideoContainer/VideoContainer";

afterEach(() => {
  cleanup();
});

test("check render correctly <LandingHeroContainer /> component", () => {
  const tree = renderWithLightTheme(<VideoContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
