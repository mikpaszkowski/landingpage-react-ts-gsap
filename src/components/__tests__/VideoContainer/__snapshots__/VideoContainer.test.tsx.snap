// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <LandingHeroContainer /> component 1`] = `
.c5 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 13rem;
  height: 13rem;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  cursor: pointer;
  background-color: rgba(254,244,230,0.8);
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  border-radius: 50%;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  will-change: transform;
}

.c5:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.2);
  background-color: rgba(254,244,230,1);
}

.c6 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 5.8rem;
  height: 5.8rem;
}

.c7 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin-left: 0.35rem;
}

.c8 {
  position: absolute;
  display: none;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin-left: 0.35rem;
}

.c3 {
  position: relative;
  margin: 0 auto;
  margin-bottom: 15rem;
  margin: 0 10rem;
  text-align: center;
  border-radius: 0.8rem;
  background: linear-gradient( 180deg, rgba(0,0,0,0.26) 0%, rgba(0,0,0,0.06) 100% );
}

.c4 {
  position: relative;
  z-index: -1;
  display: block;
  height: 100%;
  width: 100%;
  border-radius: 0.8rem;
}

.c0 {
  position: relative;
  z-index: 0;
  width: 100%;
  height: 81.1rem;
  background-color: #FABD20;
  position: relative;
  overflow: visible;
  margin-bottom: 36.3rem;
}

.c1 {
  font-family: 'Rozha One',serif;
  font-size: 8rem;
  padding-top: 6.4rem;
  margin: 0 30rem 4rem 30rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;
}

.c2 {
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-align: center;
  margin: 4rem 53.5rem 5.6rem 53.5rem;
}

@media (max-width:375px) {
  .c5 {
    width: 7.8rem;
    height: 7.8rem;
  }
}

@media (max-width:375px) {
  .c6 {
    width: 3.48rem;
    height: 3.48rem;
  }
}

@media (max-width:375px) {
  .c7 {
    display: none;
  }
}

@media (max-width:375px) {
  .c8 {
    display: block;
  }
}

@media (max-width:768px) {
  .c3 {
    margin: 0 0 15rem 0;
    padding: 0;
    width: 100%;
    border-radius: 0.8rem;
  }
}

@media (max-width:1110px) {
  .c0 {
    height: 43.7rem;
  }
}

@media (max-width:768px) {
  .c0 {
    height: 50rem;
  }
}

@media (max-width:525px) {
  .c0 {
    padding: 0 2.4rem;
    margin-bottom: 13.7rem;
    height: 49.7rem;
  }
}

@media (max-width:1110px) {
  .c1 {
    font-size: 6rem;
    text-align: center;
    margin: 0 8rem 3.7rem 8rem;
  }
}

@media (max-width:525px) {
  .c1 {
    font-size: 5rem;
    text-align: left;
    margin: 0;
    margin-bottom: 3.7rem;
  }
}

@media (max-width:1440px) {
  .c2 {
    font-size: 2rem;
    text-align: center;
    margin: 0 15rem 4rem 15rem;
    margin-bottom: 7.5rem;
  }
}

@media (max-width:525px) {
  .c2 {
    text-align: left;
    margin: 0;
    margin-bottom: 7.5rem;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    What our community is saying?
  </h1>
  <p
    className="c2"
  >
    A mission-driven company that invest in and builds healthier living
  </p>
  <div
    className="c3"
  >
    <img
      alt="Woman as freelance worker does yoga exersice in jumper."
      className="c4"
      src="/yoga-woman-image.png"
    />
    <div
      className="c5"
    >
      <div
        className="c6"
      >
        <svg
          className="c7"
        >
          playButtonPolygon.svg
        </svg>
      </div>
      <svg
        className="c8"
      >
        playButtonPolygonSmall.svg
      </svg>
    </div>
  </div>
</div>
`;
